junebug/tests/test_channel.py

```diff
@@ -72,6 +91,43 @@ def test_update_channel(self):
         self.assertEqual(update, expected)
 
     @inlineCallbacks
+    def test_update_channel_restart_transport_on_config_change(self):
+        channel = yield self.create_channel(
+            self.service, self.redis, TelnetServerTransport)
+        worker1 = channel.transport_worker
+
+        self.assertEqual(self.service.namedServices[channel.id], worker1)
+        yield channel.update({'foo': 'bar'})
+        self.assertEqual(self.service.namedServices[channel.id], worker1)
+
+        config = self.create_channel_config()
+        config['config']['foo'] = ['bar']
+        yield channel.update(config)
+
+        worker2 = channel.transport_worker
+        self.assertEqual(self.service.namedServices[channel.id], worker2)
+        self.assertNotEqual(worker1, worker2)
+
+    @inlineCallbacks
+    def test_update_channel_restart_application_on_config_change(self):
+        channel = yield self.create_channel(
+            self.service, self.redis, TelnetServerTransport)
+        worker1 = channel.application_worker
+        id = channel.application_id
+
+        self.assertEqual(self.service.namedServices[id], worker1)
+        yield channel.update({'foo': 'bar'})
+        self.assertEqual(self.service.namedServices[id], worker1)
+
+        config = self.create_channel_config()
+        config['config']['foo'] = ['bar']
+        yield channel.update(config)
+
+        worker2 = channel.application_worker
+        self.assertEqual(self.service.namedServices[id], worker2)
+        self.assertNotEqual(worker1, worker2)
```

Comment:
Shouldn't we also be restarting the application if the `mo_url` attribute changes? Not only if the `config` (which is the transport config) changes.